src/core/load_balancing/lb_policy.h

```diff
@@ -116,27 +118,33 @@ class LoadBalancingPolicy : public InternallyRefCounted<LoadBalancingPolicy> {
    public:
     virtual ~MetadataInterface() = default;
 
-    //////////////////////////////////////////////////////////////////////////
-    // TODO(ctiller): DO NOT MAKE THIS A PUBLIC API YET
-    // This needs some API design to ensure we can add/remove/replace metadata
-    // keys... we're deliberately not doing so to save some time whilst
-    // cleaning up the internal metadata representation, but we should add
-    // something back before making this a public API.
-    //////////////////////////////////////////////////////////////////////////
-
-    /// Adds a key/value pair.
-    /// Does NOT take ownership of \a key or \a value.
-    /// Implementations must ensure that the key and value remain alive
-    /// until the call ends.  If desired, they may be allocated via
-    /// CallState::Alloc().
-    virtual void Add(absl::string_view key, absl::string_view value) = 0;
+    virtual absl::optional<absl::string_view> Lookup(
+        absl::string_view key, std::string* buffer) const = 0;
 
     /// Produce a vector of metadata key/value strings for tests.
     virtual std::vector<std::pair<std::string, std::string>>
     TestOnlyCopyToVector() = 0;
+  };
 
-    virtual absl::optional<absl::string_view> Lookup(
-        absl::string_view key, std::string* buffer) const = 0;
+  /// A list of metadata mutations to be returned along with a PickResult.
+  class MetadataMutations {
+   public:
+    /// Adds a key/value pair.  If the key is already present, the new
+    /// value will be appended with a comma delimiter.
+    void Add(absl::string_view key, absl::string_view value) {
+      Add(key, grpc_event_engine::experimental::Slice::FromCopiedString(value));
+    }
+    void Add(absl::string_view key,
+             grpc_event_engine::experimental::Slice value) {
+      additions_.push_back({key, std::move(value)});
+    }
+
+   private:
+    friend class MetadataMutationHandler;
+
+    absl::InlinedVector<
+        std::pair<absl::string_view, grpc_event_engine::experimental::Slice>, 3>
```

Comment:
Let's add a comment explaining why 3 (in case we ever think to change it)

Reply:
Done.